server/src/main/java/org/apache/druid/metadata/SQLMetadataSegmentManager.java

```diff
@@ -407,6 +407,55 @@ public boolean removeSegment(SegmentId segmentId)
     }
   }
 
+  @Override
+  public boolean disableSegments(String dataSource, Collection<String> segmentIds)
+  {
+    if (segmentIds.isEmpty()) {
+      return false;
+    }
+    try {
+      connector.getDBI().withHandle(handle -> {
+        Batch batch = handle.createBatch();
+        segmentIds
+            .forEach(segmentId -> batch.add(
+                StringUtils.format(
+                    "UPDATE %s SET used=false WHERE id = '%s'",
+                    getSegmentsTable(),
+                    segmentId
+                )
+            ));
+        return batch.execute();
+      });
+    }
+    catch (Exception e) {
+      log.error(e, e.toString());
+      return false;
+    }
+    return true;
+  }
+
+  @Override
+  public boolean disableSegments(String datasource, Interval interval)
+  {
+    try {
+      final int removed = connector.getDBI().withHandle(
+          handle -> handle
+              .createStatement(StringUtils.format(
+                  "UPDATE %s SET used=false WHERE start = :start and end = :end",
```

Comment:
This needs to be a range query, what's the exclusivity behavior of the start/end?

Reply:
makes sense, will change to range query. I don't have any preferences, whatever we decide for the inclusive/exclusive behaviour of interval. I suggested [start, end) in the doc comment above.

Reply:
[start, end) sounds good

Reply:
Should this not include a `datasource = :datasource`?

Reply:
It should, good catch!